runtimes/runtimes/operational-telemetry/operational-telemetry-service.ts

```diff
@@ -0,0 +1,85 @@
+import { PeriodicExportingMetricReader } from '@opentelemetry/sdk-metrics'
+import { AWSMetricExporter } from './aws-metric-exporter'
+import { OperationalTelemetry } from './operational-telemetry'
+import opentelemetry, { Attributes } from '@opentelemetry/api'
+import { NodeSDK } from '@opentelemetry/sdk-node'
+import { Resource } from '@opentelemetry/resources'
+import { ATTR_SERVICE_NAME, ATTR_SERVICE_VERSION } from '@opentelemetry/semantic-conventions/*'
+import { randomUUID } from 'crypto'
+
+export class OperationalTelemetryService implements OperationalTelemetry {
+    private customResource: Record<string, any> = {}
+    private static instance: OperationalTelemetryService
+    private SCOPE_NAME = 'language-server-runtimes'
+    private initialized = false
+
+    private constructor() {}
+
+    reportCounterMetric(counterName: string, value = 1, attributes?: Record<string, any>): void {
+        if (!this.initialized) {
+            // todo log diag error
+            return
+        }
+
+        const meter = opentelemetry.metrics.getMeter(this.SCOPE_NAME)
+        const counter = meter.createCounter(counterName)
+        counter.add(value, attributes as Attributes)
+    }
+
+    setPeriodicGauge(gaugeName: string, valueGetter: () => number, attributes?: Record<string, any>): void {
+        if (!this.initialized) {
+            // todo log diag error
+            return
+        }
+
+        const meter = opentelemetry.metrics.getMeter(this.SCOPE_NAME)
+        const memoryUsageGauge = meter.createObservableGauge(gaugeName)
+        memoryUsageGauge.addCallback(result => {
+            result.observe(valueGetter(), attributes as Attributes)
+        })
+    }
+
+    static getInstance(): OperationalTelemetryService {
+        if (!OperationalTelemetryService.instance) {
+            OperationalTelemetryService.instance = new OperationalTelemetryService()
+        }
+        return OperationalTelemetryService.instance
+    }
+
+    initialize(serviceName: string, serviceVersion: string): void {
```

Comment:
This signature gives false understanding that you can initialize this service for different name and version, however everything after the 1st call is just ignored.

Reply:
Do we need `getInstance` and ` initialize` as a separate functions on the interface?

Reply:
I've included `initialize` in the operational telemetry interface. This method should be responsible for setting up all instances needed for collecting and sending metrics. I didn't add `getInstance` to the interface because it is up to implementation how the telemetry service is created. 

I can move these parameters to `getInstance` method, what do you think?

Reply:
Do you target delayed/controlled call to set up all instances for metrics? Why do you want `initialize` in the interface explicitly?

Reply:
Okay, we can move the `initialize` logic to the `getInstance` method. Wouldn't the signature of the new `getInstance` method then give a false impression that we could have multiple instances?